cpp/tests/listener.cpp

```diff
@@ -13,13 +13,14 @@
 
 namespace {
 
+constexpr size_t MaxProgressAttempts = 50;
```

Comment:
Yes, by running the tests for hundreds of times. 10 was already _almost_ good enough but `IsAlive` would still fail every 10-20 times or so, with 50 no failures occurred.

Reply:
Yikes, in CI:

```
[  FAILED  ] 1 test, listed below:
[  FAILED  ] ListenerTest.RaiseOnError
```

I can't reproduce it locally using the same container as CI even after running hundreds of times.

Reply:
This is a timing thing right? The OS will be timeslicing the threads so it might just not be that the thread for completion has run after calling into worker->progress() `X` number of times. Aside, where are the forward progress guarantees for UCX written down (if at all?).

On CI it is likely that the machines are more loaded than what you hvae locally.

Reply:
Yes, it could be that CI is busier and cause that indeed. I'm not aware of any forward progress guarantees for UCX. I think we could change configurations such as `UCX_KEEPALIVE_INTERVAL` to potentially force that to occur more quickly, but I don't think we want to go as far as changing default configurations just to force some test, we probably want a config that's closer to that of the user.

This would be a good test to mark flaky, but it doesn't seem to be possible doing so in vanilla gtest.

Reply:
Heres' the recommended approach https://google.github.io/googletest/advanced.html#repeating-the-tests

Reply:
I actually saw that earlier, but it isn't what we want. This is more of a replacement for the `while true; do ...; done`, it will rerun all tests, not just those that failed. It seems https://github.com/google/gtest-parallel#flakiness implements exactly what we want, but we don't use it here and I'm not sure we should add it just for that.

Reply:
If we don't want gtest parallel could we  rename the (one?) flaky test as `XXXFlaky` and run the test suite like so:

```
# The tests that are all fine
./UCXX_TESTS --gtest_filter=-*Flaky*
./UCXX_TESTS --gtest_filter=*Flaky* --repeat=10
```

Or wait, that will run until (potential) failure which doesn't do the trick :(.

Reply:
Yes, it would run until potential failure, but I think the hope is that it will succeed after one failure, so I would hope repeat 3, 5 or 10 times would be fine.

Alternatively, I could write code to run the test itself in a loop and only finally fail after `N` iterations, in which case we wouldn't need to run tests flaky/non-flaky tests separately. Essentially:

```cpp
for (size_t rep = 0; rep < N; ++rep)
{
    ...
    if (condition_is_good == true) break;
}
ASSERT_THAT(condition_is_good, true);
```

What do you think would be better?

Reply:
Let's try the loop inside the test approach, that seems easier to tune.